client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -116,7 +117,11 @@ public void onStart() {
 
     private void sendShuffleData(String taskId, List<ShuffleBlockInfo> shuffleDataInfoList) {
       try {
-        SendShuffleDataResult result = shuffleWriteClient.sendShuffleData(id.get(), shuffleDataInfoList);
+        SendShuffleDataResult result = shuffleWriteClient.sendShuffleData(
+            id.get(),
+            shuffleDataInfoList,
+            () -> isValidTaskId(taskId)
+        );
         putBlockId(taskToSuccessBlockIds, taskId, result.getSuccessBlockIds());
```

Comment:
Could we mark failed task here?

Reply:
It looks strange if marked here. The failed task should be marked by itself.

Reply:
We just process fail data blocks in the same place.

Reply:
If we can encapsulate the behavior of `mark`, why do we expose a method to other objects?

Reply:
Make sense.

Reply:
But not only the failed blocks should fast fail the task, other exception also will make task fail.

Reply:
It seems that only fail tasks will cause exception.

Reply:
We’d better not to do this assumption

Reply:
Could you avoid using ShuffleManager?  Maybe our code structure isn't proper, I concern that this pr will make writer and shuffle manger more coupled.

Reply:
Possible solution is to use the `supplier<taskId, boolean>` from shuffle manager in writer.  Do u think so?

Reply:
> Possible solution is to use the `supplier<taskId, boolean>` from shuffle manager in writer. Do u think so?

If we do so, the needCancelRequest should be renamed `FailureCallback` or other name.

Reply:
Sorry, I can't get you thought.

Reply:
> Sorry, I can't get you thought.

The solution is ok for me. But we should give the supplier a good name.

Reply:
`TaskFailureCallback` ?

Reply:
Have renamed to `TaskFailureCallback`